expenses-manager-frontend/src/main/frontend/src/app/add-expense-dialog/add-expense-dialog.component.ts

```diff
@@ -8,17 +13,76 @@ import { MatDialogRef } from '@angular/material/dialog';
 })
 export class AddExpenseDialogComponent implements OnInit {
 
-  constructor(private dialogRef: MatDialogRef<AddExpenseDialogComponent>) { }
+  constructor(private userService: UserService,
+    private expenseService: ExpenseService, 
+    private categoryService: CategoryService,
+    private currencyService: CurrencyService,
+    private dialogRef: MatDialogRef<AddExpenseDialogComponent>,
+  ) {}
+
+  categories: string[] = [];
+  currencies: string[] = [];
+
+  selectedCurrency="";
+  selectedCategory="";
+  selectedDate="";
+  selectedAmount=0;
+  description="";
 
   ngOnInit(): void {
+    this.categoryService.getAllCategories().subscribe({
+      next: (dbCategories) => {
+        this.categories = dbCategories;
+      },
+      error: (error) => {
+        console.error('Error fetching categories:', error);
+      }
+    })
+    
+    this.currencyService.getAllCurrencies().subscribe({
+      next: (dbCurrencies) => {
+        this.currencies = dbCurrencies;
+      },
+      error: (error) => {
+        console.error('Error fetching currencies:', error);
+      }
+    })
+    
   }
-  
-  selectedCurrency="RON";
-  categories: string[] = ["Food & Restaurants", "Car", "Subscriptions", "Entertainment", "Education", "Clothing", "Health"];
-  currencies: string[] = ["RON", "Euro", "US Dolar"];
 
   closeDialog(): void{
     this.dialogRef.close();
   }
 
+  private expense = {
+    description: this.description,
+    date: this.selectedDate,
```

Comment:
Pentru a defini interfete sugerez sa le definim o singura data intr-un fisier separat si sa le importam acolo unde avem nevoie. Un exemplu bun ar fi aici. Acest obeict de expense avand in vedere ca il folosesc doar in functia de submit, l-as adauga doar acolo, cu tipul pe care l-am definit in fisierul de interfete. 
Iar daca folosim forms, in functia de submit daca toate field-urile sunt valide, construim acest obiect si il adaugam in baza de date. (Nu cred ca e necesar acele functii de async await in acest caz, vom avea o singura functie de onSubmit)